tests/src/core/parse.js

```diff
@@ -43,6 +45,16 @@ describe('parse(content, { settings, ecmaFeatures })', function () {
     expect(parseSpy.args[0][1], 'custom parser to get parserOptions.filePath equal to the full path of the source file').to.have.property('filePath', path)
   })
 
+  it('passes with custom `parseForESLint` parser', function () {
+    const parseForESLintSpy = sinon.spy(eslintParser.parseForESLint)
```

Comment:
please avoid passing just an object into `sinon.spy`; that implicitly spies on all properties. Instead, let's pass explicit property names.

Reply:
It's a function

Reply:
i'm a bit confused; sinon is for spying on properties. what does passing the function in here do, as opposed to `sinon.spy(eslintParser, 'parseForEslint')`?

Reply:
It can spy function (wrapping the original one) directly. The tests are passing.😆

Reply:
Iiiiiii

Reply:
right, but if you followed my suggestion, you wouldn’t have to reassign it on the next lines.

Reply:
> right, but if you followed my suggestion, you wouldn’t have to reassign it on the next lines.

Seems more clear, I will change it.